### PR TITLE
do not accept REMIND switches in coupled config

### PR DESCRIPTION
**Pull request by orichters**
## Purpose of this PR

- avoid that REMIND switches are added to a coupled config file, where they do not have any effect.

```
Automated checks did not find counterparts scenario_config_coupled_ELEVATE2p3.csv:
  .cm_nash_autoconverge_lastrun, cm_rcp_scen, cm_startyear
You are not allowed to specify REMIND switches in coupled config.
If you find false positives, add them to knownColumnNames in scripts/start/readCheckScenarioConfig.R.
```

## Type of change

- [x] Bug fix 

## Checklist:

- [x] My code follows the [coding etiquette](https://github.com/remindmodel/remind/blob/develop/main.gms#L80)
- [x] I performed a self-review of my own code
- [x] I explained my changes within the PR, particularly in hard-to-understand areas
- [x] I checked that the [in-code documentation](https://github.com/remindmodel/remind/blob/develop/main.gms#L120) is up-to-date
- [x] I adjusted the reporting in [`remind2`](https://github.com/pik-piam/remind2) where it was needed
- [x] I adjusted `forbiddenColumnNames` in [readCheckScenarioConfig.R](https://github.com/remindmodel/remind/blob/develop/scripts/start/readCheckScenarioConfig.R) in case the PR leads to deprecated switches
- [x] All automated model tests pass (`FAIL 0` in the output of `make test`)
- [x] The changelog `CHANGELOG.md` [has been updated correctly](https://gitlab.pik-potsdam.de/rse/rsewiki/-/wikis/Standards-for-Writing-a-Changelog)

### CODE REVIEW

File: scripts/start/readCheckScenarioConfig.R [hunk 1]
```diff
@@ -179,11 +181,12 @@ readCheckScenarioConfig <- function(filename, remindPath = ".", testmode = FALSE
     # sort out known but forbidden names from unknown
     unknownColumnNames <- setdiff(unknownColumnNames, names(forbiddenColumnNames))
     if (length(unknownColumnNames) > 0) {
-      message("\nAutomated checks did not find counterparts in main.gms and default.cfg for these columns in ",
+      message("\nAutomated checks did not find counterparts ",
+              if (isFALSE(coupling)) "in main.gms and default.cfg for these columns in ",
```
Comment (dklein-pik) [inline scripts/start/readCheckScenarioConfig.R:185]:
```suggestion
              if (isFALSE(coupling)) "did not find counterparts in main.gms and default.cfg for these columns in ",
              if (isTRUE(coupling)) "found these REMIND switches in ",              
```

Comment (dklein-pik) [reply scripts/start/readCheckScenarioConfig.R:185]: Ah, ok. But why do then throw the warning "You are not allowed to specify REMIND switches in coupled config."?

Comment (orichters) [reply scripts/start/readCheckScenarioConfig.R:185]: Well. The subtle difference between telling the user that he used REMIND switches which he is not allowed to, and telling him that he used some switches that I don't know where they come from and tell him that he isn't allowed to use REMIND switches. Should I turn it into: "Maybe you specified REMIND switches, which you are not allowed to in coupled config."

Comment (dklein-pik) [reply scripts/start/readCheckScenarioConfig.R]: Still a bit confusing for me, but carry on, I don't want to get in the way any longer.

**Pull request merged**